controlplane/api/v1alpha1/conversion.go

```diff
@@ -65,6 +65,10 @@ func (src *RKE2ControlPlane) ConvertTo(dstRaw conversion.Hub) error {
 		dst.Spec.RemediationStrategy = restored.Spec.RemediationStrategy
 	}
 
+	if restored.Spec.ServerConfig.ExternalDatastoreSecret != nil {
```

Comment:
Do we need to do this for all the new fields that exist in v1beta1 but not in v1alpha1? Does it only apply to fields that are pointers or non-pointers as well?

Or essentially to address [this](https://github.com/rancher/cluster-api-provider-rke2/pull/695/files#diff-3a20e33cf3fcd6c202b903d645cebffce949557381a6caefb666c892a6bb09f3R607) instruction?
```go
// WARNING: in.ExternalDatastoreSecret requires manual conversion: does not exist in peer-type
```

Reply:
We do this for all new fields that are not already being converted using the (autogenerated) conversion [function](https://github.com/rancher/cluster-api-provider-rke2/blob/e1fa5a84a1f602e95aa2c937ce5c5f88999959cc/controlplane/api/v1alpha1/zz_generated.conversion.go#L331).  The warning is also a reminder that this needs to be manually taken care of.